lib/routemap.h

```diff
@@ -241,6 +241,146 @@ struct route_map {
 };
 DECLARE_QOBJ_TYPE(route_map)
 
+/* Route-map match conditions */
+#define IS_MATCH_INTERFACE(C)                                                  \
```

Comment:
This won't work as you expect:
- `strncmp` does not have the same functionality as `strcmp`
- `enum` will be more performant and will avoid all these macros

Please change this.

Reply:
Library is not ready for identity-ref, so we are using strcmp. Once the code is pushed for Identity-ref we will replace these with enum

Reply:
@patrasar Rafael is right about `strncmp`.

For example, if you pass `"frr-bgp-route-map:ip-route-source"`, both `IS_MATCH_ROUTE_SRC` and `IS_MATCH_ROUTE_SRC_PL` will return `true`.

Let's replace it with `strmatch` like it's done in other places where we use identityrefs.

Reply:
Done